client/src/components/logViewer/sasDiagnostics.ts

```diff
@@ -0,0 +1,135 @@
+// Copyright © 2023, SAS Institute Inc., Cary, NC, USA.  All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import {
+  Diagnostic,
+  DiagnosticCollection,
+  DiagnosticSeverity,
+  Range,
+  Uri,
+  languages,
+} from "vscode";
+
+import { LogLine, OnLogFn } from "../../connection";
+import { SASCodeDocument } from "../utils/SASCodeDocument";
+import { isShowProblemsFromSASLogEnabled } from "../utils/settings";
+import { Problem, parseLog } from "./logParser";
+
+let diagnosticCollection: DiagnosticCollection;
+
+enum DiagnosticCommands {
+  IgnoreCommand = "SAS.diagnostic.ignore",
+  IgnoreAllWarningCommand = "SAS.diagnostic.ignoreAllWarning",
+  IgnoreAllErrorCommand = "SAS.diagnostic.ignoreAllError",
+  IgnoreAllCommand = "SAS.diagnostic.ignoreAll",
+}
+
+function ignore(diagnosticsToRemove: Diagnostic[], uri: Uri): void {
+  const diagnostics = getSasDiagnosticCollection().get(uri);
+  const newDiagnostics = diagnostics.filter((diagnostic) => {
+    return !diagnosticsToRemove.includes(diagnostic);
+  });
+
+  getSasDiagnosticCollection().set(uri, newDiagnostics);
+}
+
+function ignoreAll(uri: Uri, severity?: DiagnosticSeverity): void {
+  if (severity === undefined) {
+    getSasDiagnosticCollection().delete(uri);
+  } else {
+    const diagnostics = getSasDiagnosticCollection().get(uri);
+    const newDiagnostics = diagnostics.filter((diagnostic) => {
+      return diagnostic.severity !== severity;
+    });
+    getSasDiagnosticCollection().set(uri, newDiagnostics);
+  }
+}
+
+function getSasDiagnosticCollection(): DiagnosticCollection {
+  if (diagnosticCollection === undefined) {
+    diagnosticCollection = languages.createDiagnosticCollection("sas");
+  }
+  return diagnosticCollection;
+}
+
+async function updateDiagnostics(
+  logs: LogLine[],
+  codeDoc: SASCodeDocument,
+): Promise<void> {
+  if (!isShowProblemsFromSASLogEnabled()) {
+    return;
```

Comment:
Disabling this setting should clear existing diagnostices if any

Reply:
Done